app/utilities/protocol_importers/protocol_intermediate_object.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module ProtocolImporters
+  class ProtocolIntermediateObject
+    attr_accessor :normalized_protocol_data, :user, :team, :protocol
+
+    def initialize(normalized_json: {}, user:, team:)
+      @normalized_protocol_data = normalized_json.with_indifferent_access[:protocol] if normalized_json
+      @user = user
+      @team = team
+    end
+
+    def import
+      build unless @protocol
+      @protocol.save
+      @protocol
+    end
+
+    def build
+      @protocol = Protocol.new(protocol_attributes)
+      @protocol.description = ProtocolDescriptionBuilder.generate(@normalized_protocol_data&.reject { |k| k == :steps })
```

Comment:
We could let `ProtocolDescriptionBuilder` handle that rejection, easier for the brain.

Reply:
We do not need steps in `ProtocolDescriptionBuilder`. and there could be many steps. Do we care about sending JSON without steps there? (steps array could be long)

Reply:
I was just thinking from the perspective of classes. If you want to use `ProtocolDescriptionBuilder`  then you always have to have in mind to exclude steps (although I'm not sure if this breaks something). If you think it's ok as it is, I trust you completely! Just a thought.